docs/howtos/extending/magicgui.md

```diff
@@ -754,29 +762,63 @@ simply provide the class definition and add to the plugin manifest.
 ### `QtWidgets.QWidget`
 
 For the most control over your widget, subclass
-[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html):
+[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html).
+In the following example, we create a button and a dropdown list.
+The available choices for the dropdown list are the
+current layers in the viewer. For this, we use
+{func}`create_widget <magicgui.widgets.create_widget>` with the annotation
+{attr}`napari.types.ImageData`.
+In opposition to
+using `@magicgui` as [shown above](#parameter-annotations), we now need to
```

Comment:
I do not understand that comment. Can you elaborate more on this?

Reply:
Ah sorry, I wasn't clear and it's difficult to word this part as its complicated.

Whenever you use a `magicgui` `ContainerWidget`, type annotating or `create_widget` widget works without any more work. So this means e.g.,
* the `@magicgui` decorator,
* subclassing `ContainerWidget`,
* subclassing `FunctionGui` (see https://pyapp-kit.github.io/magicgui/widgets/#the-widget-hierarchy).

(not just `@magicgui` as we've said in the text).

I guess the #parameter-annotations link is also awkward, because it only talks about using type hints in `@magicgui` but not in `create_widgets`. I think this page itself just needs work. I'm happy to leave the link as is and we can think about improving this page later.

Reply:
I think it would be clearest here to say the following:

```suggestion
Because the layer selection widget will be housed by a native `QWidget`
and not by a `magicgui` subclass as [shown above](#parameter-annotations), we now need to
```

Reply:
Now I understand. Yes, that is an important point that I was not aware of / thinking of.